### ISSUE TITLE
Composition RenderNode prefabs

### ISSUE THREAD

**Issue by Aeva** [Aeva/m.grl#101]
Once issue #100 is complete, it will be possible to have interesting pefabs for rendering.  Issue #95 should be implemented as a RenderNode prefab.  This would, however, not be utilized in the user-defined rendering stack, but rather would make use of an early pipeline stage.

This issue focuses on composition RenderNodes, which take two image buffers and interpolates between them.  This can be used for stereoscopic rendering modes, transition effects (eg screen wipe, shatter, etc), as well as for loading screens.

 - [x] make the stereoscopic demo's fragment shader to be a bundled shader
 - [ ] ~~rendernode for stereoscopic rendering~~ (moving into own issue)
 - [x] simple screen wipe transition effect
 - [x] basic loading screen prefab
 - [x] show loading progress in loading screen prefab

The loading screen prefab includes a render node and a simple bundled image or three to visualize a the loading bar etc.  When the media ready event is fired, it triggers a transition effect (something that could be swapped out via a property on the loading screen render node).  The loading screen rendernode probably should have a .reset_loading method, so that it can be activated again at a later point.

It would be a reasonable configuration to have it so that the loading screen render node is always the top level render node in the tree, and it should be optimized such that this does not introduce any significant overhead.

It could also be that the loading screen itself is a separate render node that can be overridden should the dev desire to do so.  Different 'themes' might be bundled with m.grl so that one can have some basic customization in place regardless.

**Comment by Aeva** [comment]
Most of the GLSL can be pulled from existing demos now.  The loading screen prefab would need to have a bundled image or three.

Transition effects should probably work in such a way that you can call a "blend_to" method on them (rendernode and time as an arg) to cue up a wipe; the peek method would be used on a transition effect to proxy out to another node when the animation is not active.  This part can probably just be implemented as a special constructor that inherits from RenderNode, which can be instance with this or that shader, and assumes that the shader has a particular uniform var for controlling the effect.

The loading screen prefab would just be a render node that shows the loading progress in an appealing way, and then one manually calls a wipe when the loading sequence finishes.  Otherwise, the code gets kind of hairy with regards to the order of events getting scheduled (eg, you probably build your scene graph after your resources load heh and thus you don't have the full render pipeline constructed right away)

**Comment by Aeva** [comment]
There is a "TransitionEffect" constructor now, so one can easily define scene wipe effects.  It would be nice to have a "transition complete" callback, though.

At any rate, it should be easy to bundle a few basic wipe shaders etc.

**Comment by Aeva** [comment]
As for the loading screen prefab; it may be a good idea to have some kind of onload callback for it so that people don't strictly need to attach to mgrl_on_media_ready?  Will need to think more on this.

**Comment by Aeva** [comment]
The loading screen prefab could benefit from the dom management system described in #135.  This would allow for managing the progress text rather than requiring any applications to do so.

**Comment by Aeva** [comment]
As VR is somewhat lower priority right now due to present technical limitations and low availability of hardware, it doesn't make sense for stereoscopic rendering to hold up 1.0.  As such, that requirement for this ticket has been moved off into its own ticket #170
